spot_rl_experiments/spot_rl/envs/skill_manager.py

```diff
@@ -35,10 +41,14 @@ def __init__(self):
         # Create a local waypoint dictionary
         self.waypoints_yaml_dict = get_waypoint_yaml()
 
+        # Flag will decide whether we want to shutdown or keep the spot same
+        self.shutdownndock_on_delete = shutdownndock_on_delete
+
     def __del__(self):
-        self.spot.shutdown(should_dock=True)
+        if self.shutdownndock_on_delete:
+            self.spot.shutdown(should_dock=True)
```

Comment:
This is redundant.

If someone is done using spot, they can just call either of the following

1. `skill_manager.dock()` which does call shutdown internally
2. spot.shutdown() as a direct call to shutdown the robot.

This will ensure user is calling shutdown on its own rather than shutting down in destructors and creating extra unforeseen errors.

Reply:
Yes maybe if the original branch just overrides the __del__ method in the class, I can remove this flag